app/views/browse/mobile-clients.html

```diff
@@ -0,0 +1,57 @@
+<div class="middle">
+  <div class="middle-header">
+    <div class="container-fluid">
+      <breadcrumbs breadcrumbs="breadcrumbs"></breadcrumbs>
+      <alerts alerts="alerts"></alerts>
+      <div>
+        <h1 class="contains-actions">
+          <div class="pull-right dropdown" ng-if="mobileClient">
+            <button type="button" class="dropdown-toggle btn btn-default actions-dropdown-btn hidden-xs" data-toggle="dropdown">
+              Actions
+              <span class="caret" aria-hidden="true"></span>
+            </button>
+            <a href=""
+               class="dropdown-toggle actions-dropdown-kebab visible-xs-inline"
+               data-toggle="dropdown"><i class="fa fa-ellipsis-v"></i><span class="sr-only">Actions</span></a>
+            <ul class="dropdown-menu dropdown-menu-right actions action-button">
+              <!--TODO add canI filter, needs an update to AuthorisationService-->
+              <li>
+                <delete-link
+                  kind="MobileClient"
+                  group="mobile.k8s.io"
+                  resource-name="{{mobileClient.metadata.name}}"
+                  project-name="{{mobileClient.metadata.namespace}}"
+                  alerts="alerts"
+                  redirect-url={{redirectUrl}}>
+                </delete-link>
+              </li>
+            </ul>
+          </div>
+          {{mobileClient.spec.name}}
```

Comment:
You pass it to $scope. If mobileClient.spec.name is undefined you can use default value and this one should have meaningful name rather then name taken from k8s